### PR TITLE
add `gliner`

### PR DESCRIPTION
**Pull request by moritzwilksch**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by moritzwilksch** [comment]
win builds failing due to missing pytorch
```
conda_libmamba_solver.conda_build_exceptions.ExplainedDependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {MatchSpec(">=1.5.0,!=1.8"), MatchSpec("python-flair==0.13.1=pyhd8ed1ab_0"), MatchSpec("pytorch[version='>=2.0.0']")}
Encountered problems while solving:
  - nothing provides pytorch >=1.5.0,!=1.8 needed by python-flair-0.13.1-pyhd8ed1ab_0
  - nothing provides requested pytorch >=2.0.0
```

**Comment by moritzwilksch** [comment]
@conda-forge/help-python, ready for review!

File: recipes/gliner/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/gliner-{{ version }}.tar.gz
+  sha256: 2c00570066a016d3ec682fc0d1aa0c61e7318873343662233c6994ac37fb2a10
+
+build:
+  noarch: python  # won't work on windows due to missing pytorch win build
```
Comment (ocefpaf) [inline recipes/gliner/meta.yaml:13]: No need for this comment. The recipe and the artifact will work if an end user can provide pytorch for Windows vai another channel. That is the beauty of noarch packages.

```suggestion
  noarch: python
```

File: recipes/gliner/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  run:
+    - python >=3.8
+    - pytorch >=2.0.0
+    - transformers >=4.38.2
+    - huggingface_hub >=0.21.4
+    - python-flair >=0.13.1
+    - scipy ==1.12
+    - seqeval
+    - tqdm
+    - urllib3 <2.0.0,>=1.0.0  # inherited from flair
```
Comment (ocefpaf) [inline recipes/gliner/meta.yaml:31]: What do you mean "inherited from flair"? Is it an extra dependency of flair that this package requires?

Comment (moritzwilksch) [reply recipes/gliner/meta.yaml:31]: yes, flair seems to require this pin as it won't build without it. Could also just be that grayskull didn't pick this up in which case we can just remove the comment.

Comment (ocefpaf) [reply recipes/gliner/meta.yaml:31]: You can leave it there. I was just trying to understand what was going on.

**Pull request merged**